master/internal/core.go

```diff
@@ -1090,18 +1090,31 @@ func buildRM(
 	// Set the default RM name for the multi-rm, from the default RM index.
 	defaultRMName := rmConfigs[config.DefaultRMIndex].ResourceManager.Name()
 	rms := map[string]rm.ResourceManager{}
+	clusterNames := map[string]int{}
 
 	for _, cfg := range rmConfigs {
 		c := cfg.ResourceManager
 		switch {
 		case c.AgentRM != nil:
+			rmClusterName := c.AgentRM.ClusterName
+			if len(rmClusterName) == 0 {
+				return nil, fmt.Errorf("resource manager %s must have a cluster name: %w", c.Name())
+			}
```

Comment:
still not understanding why we need to require ClusterName for a cluster when we already require the 'Name' field for the RM?

Reply:
Ref [above reply](https://github.com/determined-ai/determined/pull/9226/files#r1603680676).